src/simplification/Simplifier.ts

```diff
@@ -64,14 +65,19 @@ export class Simplifier {
 
     if (typeof schema !== 'boolean') {
       this.seenSchemas.set(schema, model);
-      //All schemas of type object MUST have ids, for now lets make it simple
+      // All schemas of type object MUST have ids, for now lets make it simple
       if (model.type !== undefined && model.type.includes('object')) {
         const schemaId = schema.$id ? schema.$id : `anonymSchema${this.anonymCounter++}`;
         model.$id = schemaId;
       } else if (schema.$id !== undefined) {
         model.$id = schema.$id;
       }
 
+      const name = simplifyName(schema, model.$id);
+      if (name) {
+        model.name = name;
+      }
+
```

Comment:
What is the different between model name and model id? 🤔 Why have both?

Reply:
> Does it make sense to have $id as part of the CommonModel anymore since your new name kinda replaces this right? 

We had a discussion and as I remember we decided that we should have `$id` as a separate field and name should be written in a wrapper of Schema (which I didn't do because it creates an additional wrapper for one field and it creates not needed complexity).

> This also means you can remove the anonymous schema id naming functionality.

Nope, anonymous is needed, when you have allOf, oneOf etc schemas -> without title or separate extension for naming you cannot name this schema.

Reply:
🤔 The reason why it should be moved to another class wrapper for Schema was because we  could keep our originalSchema as is. IMO this naming functionality should completely remove our current very basic anonymous naming including allOf/oneOf etc and infer naming of models more correctly. I dont see why we need to have both?

Reply:
> IMO this naming functionality should completely remove our current very basic anonymous naming including allOf/oneOf etc and infer naming of models more correctly. I dont see why we need to have both?

How do you want to avoid anonymSchema when you have an object model inside oneOf, and then infer this model to `models` array?

Reply:
One example is to use the path as the name - similar to what is suggested in https://github.com/asyncapi/parser-js/issues/176 

Pretty sure we discussed this as well where I said you should be careful about using the property names alone as names. I actually thought I made an issue here in the rep but I guess we took it in slack 🤔